File: dynamicDebug/src/main/java/com/cloud/dynamicdebug/MockFileModule.java
package com.cloud.dynamicdebug;

import com.cloud.disk.api.file.TransferFile;

import dagger.Binds;
import dagger.Module;
import dagger.hilt.InstallIn;
import dagger.hilt.android.components.ActivityComponent;

@Module
@InstallIn(ActivityComponent.class)
public abstract class MockFileModule {
    @Binds
    public abstract TransferFile bindTransferFile(
            MockTransferFileImpl transferFileImpl
    );
}

File: dynamicDebug/src/main/java/com/cloud/dynamicdebug/MockFileModule.kt
package com.cloud.dynamicdebug

import com.cloud.disk.api.file.TransferFile
import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent

@Module
@InstallIn(SingletonComponent::class)
abstract class MockFileModule {
    @Binds
    abstract fun bindTransferFile(transferFileImpl: MockTransferFileImpl): TransferFile?
}